Launcher/src/windows/update.c

```diff
@@ -0,0 +1,116 @@
+#include <archive.h>
+#include <archive_entry.h>
+#include <assert.h>
+#include <dirent.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <windows.h>
+
+#define BUFFER_SIZE (1024 * 1024 * 16)
+#define DELIM "\\"
+
+int copyData(struct archive *in, struct archive *out) {
+	const void *buff;
+	size_t size;
+	la_int64_t offset;
+	while (1) {
+		int result = archive_read_data_block(in, &buff, &size, &offset);
+		if (result == ARCHIVE_EOF)
+			return ARCHIVE_OK;
+		else if (result < ARCHIVE_OK)
+			return result;
+		result = archive_write_data_block(out, buff, size, offset);
+		if (result < ARCHIVE_OK) {
+			return result;
+		}
+	}
+	return ARCHIVE_OK;
+}
+
+void launcher_update(char* baseDir) {
+	assert(baseDir);
+	char *updateDir = malloc(strlen(baseDir) + 3 + 9);
+	assert(updateDir);
+	sprintf(updateDir, "%s%s%s", baseDir, DELIM, "updates");
+	char *updateFile = malloc(strlen(updateDir) + 3 + 10);
+	assert(updateFile);
+	sprintf(updateFile, "%s%s%s", updateDir, DELIM, "data.zip");
+
+	struct stat s;
+	if (stat(updateFile, &s) != 0) {
+		printf("No update-file found in %s\n", updateDir);
```

Comment:
On windows nobody will see this print statements. Such a thing should be shown graphically. (E.g. replace the progressbar of downloading/extracting Java with a label that will show these messages)